### PR TITLE
Reject invalid messages from incoming connections - Closes #4424

### LINKED ISSUE #4424: Reject unnecessary messages and ban a peer
### Expected behavior

A node should not send internal messages to a lisk node as it's not needed which creates an extra overhead.

### Actual behavior

Its accepting messages that are internal and not related to `lisk-sdk` and should be rejected and a peer should be banned as its not suppose to send these messages.

### Which version(s) does this affect? (Environment, OS, etc...)
`3.0`

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

Peers can send invalid messages and overwhelm a node

### How did I solve it?

Listen to invalid messages and terminate the connection

### Review checklist

- [x] The PR resolves #4424 
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)

### CODE REVIEW

File: elements/lisk-p2p/src/p2p.ts [hunk 1]
```diff
@@ -640,8 +640,30 @@ export class P2P extends EventEmitter {
 
 	private async _startPeerServer(): Promise<void> {
 		this._scServer.on(
-			'connection',
+			'handshake',
 			(socket: SCServerSocket): void => {
+				// Terminate the connection the moment it receive ping frame
+				(socket as any).socket.on('ping', () => {
```
Comment (shuse2) [inline elements/lisk-p2p/src/p2p.ts]: wouldn't this happen on the normal scenario? smme for `pong`

File: elements/lisk-p2p/test/integration/reject_unwanted_messages.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+import { P2P, EVENT_INBOUND_SOCKET_ERROR } from '../../src/index';
+import {
+	createNetwork,
+	destroyNetwork,
+	SEED_PEER_IP,
+} from 'utils/network_setup';
+import { wait } from 'utils/helpers';
+import expect = require('expect');
+const INVALID_MESSAGE_1 = '#1';
```
Comment (diego-G) [inline elements/lisk-p2p/test/integration/reject_unwanted_messages.ts]: Reuse `SOCKET_PING_MESSAGE` or change the content.

File: elements/lisk-p2p/src/peer/base.ts [hunk 3]
```diff
@@ -201,7 +209,28 @@ export class Peer extends EventEmitter {
 			this.emit(EVENT_REQUEST_RECEIVED, request);
 		};
 
-		this._handleWSMessage = () => {
+		this._handleWSMessage = (message: any) => {
```
Comment (diego-G) [inline elements/lisk-p2p/src/peer/base.ts]: Please cover this with unit tests.

File: elements/lisk-p2p/test/integration/reject_unwanted_messages.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+		await wait(500);
+	});
+
+	afterEach(async () => {
+		await destroyNetwork(p2pNodeList);
+	});
+
+	it('should set the isActive property to true for all nodes', async () => {
```
Comment (diego-G) [inline elements/lisk-p2p/test/integration/reject_unwanted_messages.ts]: Change the description to something more meaningful with regard to the current expectations.

File: elements/lisk-p2p/src/p2p.ts (hunk 1 above)
Comment (ishantiw) [reply elements/lisk-p2p/src/p2p.ts]: These low level `ping` messages are not supposed to be sent by lisk nodes so it should be fine to disconnect with them as soon as we receive it.

File: elements/lisk-p2p/src/peer/base.ts (hunk 3 above)
Comment (ishantiw) [reply elements/lisk-p2p/src/peer/base.ts]: It's covered in the integration test as we require 2 nodes to test it properly

Comment (shuse2) [inline elements/lisk-p2p/src/peer/base.ts]: This change is no longer needed

**Pull request merged**